### PR TITLE
Internal: Editor V2 - Publish button [ED-9656]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

File: packages/packages/documents/src/sync/__tests__/test-utils.ts [hunk 1]
```diff
@@ -60,18 +61,18 @@ export function makeMockV1Document( id = 1 ): V1Document {
...
 				post_title: 'Document ' + id,
 				post_status: 'publish',
-			} ) as V1Document['container']['settings'],
+			} ),
 		},
 	};
 }
 
-function makeSettings<T extends object>( settings: T ) {
+export function makeV1Settings<T extends object>( settings: T ) {
```
Comment (rotemee) [inline packages/packages/documents/src/sync/__tests__/test-utils.ts:73]: Add comment.

**Review by rotemee — CHANGES_REQUESTED** [review]

**Review by rotemee — DISMISSED** [review]

File: packages/packages/documents/src/sync/utils.ts [hunk 2]
```diff
@@ -11,11 +11,17 @@ export function getV1DocumentsManager() {
 }
 
 export function normalizeV1Document( documentData: V1Document ): Document {
+	// TODO: See: https://github.com/elementor/elementor/blob/e86b957a2/assets/dev/js/editor/document/save/behaviors/footer-saver.js#L96-L101
+	const status = documentData.container.settings.get( 'post_status' ) ?? 'pending';
+
+	const isAutoSave = documentData.config.revisions.current_id !== documentData.id;
```
Comment (Nevoss) [inline packages/packages/documents/src/sync/utils.ts:17]:
```suggestion
	const isUnpublishedRevision = documentData.config.revisions.current_id !== documentData.id;
```

with comment of draft and autosave

File: packages/packages/documents/src/components/top-bar/__tests__/primary-action.test.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		} );
+
+		// Act.
+		const { getByRole } = render( <PrimaryAction /> );
+
+		// Assert.
+		expect( getByRole( 'button' ) ).toBeEnabled();
+	} );
+
+	it( 'should save the current document on click', () => {
```
Comment (Nevoss) [inline packages/packages/documents/src/components/top-bar/__tests__/primary-action.test.tsx:123]:
```suggestion
	it( 'should save the active document on click', () => {
```

**Review by Nevoss — CHANGES_REQUESTED** [review]

File: packages/packages/documents/src/components/top-bar/primary-action.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+export default function PrimaryAction() {
+	const document = useActiveDocument();
+	const { save } = useActiveDocumentActions();
+
+	if ( ! document ) {
+		return null;
+	}
+
+	const isDisabled = ! isEnabled( document );
+	const shouldShowSpinner = document.isSaving && ! isDisabled;
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply packages/packages/documents/src/components/top-bar/primary-action.tsx:15]: Done

File: packages/packages/documents/src/hooks/__tests__/use-active-document.test.tsx [hunk 5]
```diff
@@ -24,6 +24,7 @@ describe( '@elementor/documents - useActiveDocument', () => {
 			id: 1,
 			title: 'Document 1',
 			status: 'publish' as const,
+			type: 'page',
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply packages/packages/documents/src/hooks/__tests__/use-active-document.test.tsx:27]: Done

File: packages/packages/documents/src/sync/utils.ts (hunk 2 above)
Comment (StyleShit) [reply packages/packages/documents/src/sync/utils.ts:17]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/primary-action.test.tsx (hunk 3 above)
Comment (StyleShit) [reply packages/packages/documents/src/components/top-bar/__tests__/primary-action.test.tsx:123]: Done

**Review by Nevoss — DISMISSED** [review]

**Review by Nevoss — APPROVED** [review]

**Pull request merged**